### PR TITLE
fix: add checks in placese show by crashes reported on sentry

### PR DESCRIPTION
**Pull request by summeroff**

### CODE REVIEW

File: deps/media-playback/media-playback/media.c [hunk 1]
```diff
@@ -293,7 +295,8 @@ static void mp_media_next_audio(mp_media_t *m)
 
 		if (audio->format == AUDIO_FORMAT_UNKNOWN) {
 			for (size_t j = 0; j < MAX_AV_PLANES; j++) {
-				free(audio->data[j]);
+				if(audio->data[j]!=NULL)
```
Comment (eddyStreamlabs) [inline deps/media-playback/media-playback/media.c]: Style issue here: `if(audio->data[j] != NULL)` would be better I think

File: libobs/obs-source.c [hunk 2]
```diff
@@ -2566,7 +2565,8 @@ static void copy_audio_data(obs_source_t *source,
 			source->audio_data.data[i] = bmalloc(size);
 		}
 
-		memcpy(source->audio_data.data[i], data[i], size);
+		if(data[i]!=NULL)
```
Comment (eddyStreamlabs) [inline libobs/obs-source.c]: Style issue here: `if(data[i] != NULL)` would be better I think

File: libobs/obs-source.c [hunk 3]
```diff
@@ -2417,8 +2417,7 @@ void obs_source_output_video(obs_source_t *source,
 		return;
 	}
 
-	struct obs_source_frame *output = !!frame ?
-		cache_video(source, frame) : NULL;
+	struct obs_source_frame *output = cache_video(source, frame);
```
Comment (eddyStreamlabs) [inline libobs/obs-source.c]: Was removing the check intentional?

Comment (summeroff) [reply libobs/obs-source.c]: yes. value of frame is get checked few lines above so checking it again is misleading.

Comment (eddyStreamlabs) [reply libobs/obs-source.c]: got it

**Pull request merged**